### ISSUE TITLE
Incorrect Town tileset during Winter (Client Only)

### ISSUE THREAD

**Issue by llamasoft** [Pathoschild/SMAPI#552]
SMAPI causes multiplayer clients to load the incorrect Town tileset in Winter.  When the multiplayer client is running a vanilla install (without SMAPI) it loads the correct Winter tileset, but with SMAPI it incorrectly loads the Spring tileset but continues to use the Winter sprites and effects.  The issue only affects multiplayer clients, not the host.

This is in regards to this forum post: https://community.playstarbound.com/threads/stardew-valley-multiplayer-beta-known-issues-fixes-2.145034/page-23#post-3292097

**Comment by Pathoschild** [comment]
@llamasoft Does this happen during the season transition, or does it happen when loading directly into winter too?

**Comment by llamasoft** [comment]
[SMAPI-latest.txt](https://github.com/Pathoschild/SMAPI/files/2128174/SMAPI-latest.txt)
[SMAPI-latest.player-2.txt](https://github.com/Pathoschild/SMAPI/files/2128175/SMAPI-latest.player-2.txt)

I first encountered it organically via season transition.  However, while debugging, I made a separate file and forced the season to Winter with `world_setseason winter` and was able to reproduce the issue.

Attached are the host and client (`.player-2`) logs.  The only thing that catches my attention is that the client warping to town also triggers a `Context: set location to BusStop`.

Host logs:

```
[11:16:27 TRACE SMAPI] Context: set location to Farm.
[11:16:28 TRACE game] Lobby entered: 123385157199481766
[11:16:28 TRACE game] Steam lobby entered: 109775244988572904
[11:16:30 TRACE game] Warping to BusStop
[11:16:31 TRACE SMAPI] Context: set location to BusStop.
[11:16:34 TRACE game] Warping to BusStop
[11:16:41 TRACE game] Warping to Town
[11:16:42 TRACE SMAPI] Context: set location to Town.
```

Client logs:

```
[11:17:16 TRACE game] Warping to Farm
[11:17:16 TRACE SMAPI] Context: set location to Farm.
[11:17:23 TRACE game] Warping to BusStop
[11:17:23 TRACE SMAPI] Context: set location to BusStop.
[11:17:31 TRACE game] Warping to Town
[11:17:31 TRACE SMAPI] Context: set location to BusStop.   <--- This seems out of place
[11:17:31 TRACE SMAPI] Context: set location to Town.
```

**Comment by llamasoft** [comment]
I stand corrected regarding the double Context.  It doesn't appear to matter how one comes to arrive at Town.  Using CJB Cheats Menu resulted in a double Context, but using the console `debug warp` commands resulted in a single Context.

I tried warping from different locations to town using CJB Cheats and `debug` commands, but the result was still the incorrect Spring appearance.  I also tried warping from Town to itself just to see if it was an edge case, but no luck there either.
